ntbs-service/DataAccess/DataQualityRepository.cs

```diff
@@ -106,24 +106,31 @@ public async Task<IList<Notification>> GetNotificationsEligibleForDotVotAlerts()
+            return await _context.Notification
+                .SelectMany(_ => _context.Notification, 
                     (notification, duplicate) => new {notification, duplicate})
                 .Where(t =>
+                    // Check that one of the following cases are true:
+                    // The notification's names match directly (and it is not matching on itself)
+                    (t.notification.PatientDetails.GivenName == t.duplicate.PatientDetails.GivenName &&
+                     t.notification.PatientDetails.FamilyName == t.duplicate.PatientDetails.FamilyName &&
+                     t.notification.NotificationId != t.duplicate.NotificationId)
+                    ||
+                    // The notification's given and family names match but are reversed
                     (t.notification.PatientDetails.GivenName == t.duplicate.PatientDetails.FamilyName &&
-                     t.notification.PatientDetails.FamilyName == t.duplicate.PatientDetails.GivenName &&
-                     t.notification.PatientDetails.NhsNumber == t.duplicate.PatientDetails.NhsNumber &&
-                     t.notification.PatientDetails.NhsNumber != null &&
-                     t.notification.PatientDetails.GivenName != null &&
-                     t.notification.PatientDetails.GivenName != null) 
+                     t.notification.PatientDetails.FamilyName == t.duplicate.PatientDetails.GivenName)
                     ||
-                    (t.notification.PatientDetails.NhsNumber == t.duplicate.PatientDetails.NhsNumber &&
-                     t.notification.PatientDetails.Dob == t.duplicate.PatientDetails.Dob && 
-                     t.notification.PatientDetails.NhsNumber != null &&
-                     t.notification.PatientDetails.Dob != null)
+                    // The notification's date of births match
+                    (t.notification.PatientDetails.Dob == t.duplicate.PatientDetails.Dob)
                     &&
+                    // Then check that the notification's nhs numbers match, the notification is notified and the notifications
+                    // have been notified for at least the minimum amount of time specified
+                    t.notification.PatientDetails.NhsNumber == t.duplicate.PatientDetails.NhsNumber &&
                     t.notification.NotificationDate < DateTime.Now.AddDays(-DataQualityPotentialDuplicateAlert.MinNumberDaysNotifiedForAlert) &&
-                    t.duplicate.NotificationDate < DateTime.Now.AddDays(-DataQualityPotentialDuplicateAlert.MinNumberDaysNotifiedForAlert))
+                    t.duplicate.NotificationDate < DateTime.Now.AddDays(-DataQualityPotentialDuplicateAlert.MinNumberDaysNotifiedForAlert) &&
+                    t.notification.NotificationStatus == NotificationStatus.Notified &&
+                    t.duplicate.NotificationStatus == NotificationStatus.Notified &&
+                    t.notification.GroupId != t.duplicate.GroupId)
```

Comment:
I'm worried that this will work wrongly for notifications where both have groupId as `null`. Null comparison works differently in csharp and in sql, so couldn't even guess which way this resolves, so even if this works correctly now, I think it would be worth dealing with nulls explicitly!